data_pipeline/monitoring/monitoring_config.py

```diff
@@ -0,0 +1,22 @@
+from data_pipeline.utils.pipeline_config import (
+    update_deployment_env_placeholder
+)
+
+
+# pylint: disable=too-few-public-methods, too-many-instance-attributes
+class MonitoringConfig:
+
+    def __init__(self, data_config: dict,
```

Comment:
This is how we have done it for other config. So it is great for consistency.
I am just putting this out there as a potential option for the future...
Perhaps consider making it a `NamedTuple`. Those seem to look neater and they for example come with a `__repr__` implementation.

Something like:

```python
class MonitoringConfig(NamedTuple):
    project_name: str
    dataset_name: str
    table_name: str
    bucket_name: str
    object_name: str

    @staticmethod
    def from_dict(data_config: dict, ...):
        resolved_data_config = update_deployment_env_placeholder(...)
        return MonitoringConfig(
            project_name=resolved_data_config.get('project_name'),
            dataset_name=resolved_data_config.get('dataset_name'),
            ...
        )
```

The other thing that might be worth considering is, what configuration options are required? I think we came across that during the gmail development, when a value was misconfigured and caused a None exception later on, rather than when loading the config. Using `data_config['project']` would at least require it to be set (not necessarily being non-empty).